### PR TITLE
chore(tests): add CommonRouteSpec mock to reduce parser tests length

### PR DESCRIPTION
**Pull request by jrsmroz**
**What this PR does / why we need it**:

**Which issue this PR fixes**:

**Special notes for your reviewer**:

**PR Readiness Checklist**:

Complete these before marking the PR as `ready to review`:

- [ ] the `CHANGELOG.md` release notes have been updated to reflect any significant (and particularly user-facing) changes introduced by this PR

### CODE REVIEW

File: internal/dataplane/parser/translate_httproute_test.go [hunk 1]
```diff
@@ -1305,3 +1245,11 @@ func TestIngressRulesFromHTTPRoutes_RegexPrefix(t *testing.T) {
 func HTTPMethodPointer(method string) *gatewayv1beta1.HTTPMethod {
 	return (*gatewayv1beta1.HTTPMethod)(&method)
 }
+
+func commonRouteSpecMock() gatewayv1beta1.CommonRouteSpec {
```
Comment (pmalek) [inline internal/dataplane/parser/translate_httproute_test.go:1249]: Maybe accept a name param?

Comment (jrsmroz) [reply internal/dataplane/parser/translate_httproute_test.go]: I changed that, but given it's a mock it makes no difference.

Comment (jrsmroz) [reply internal/dataplane/parser/translate_httproute_test.go]: Linter is not happy :D
```
Error: internal/dataplane/parser/translate_httproute_test.go:1160:26: `commonRouteSpecMock` - `parentReferentName` always receives `"fake-gateway"` (unparam)
func commonRouteSpecMock(parentReferentName string) gatewayv1beta1.CommonRouteSpec {
 ```

**Review by pmalek — APPROVED** [review]
heh, alright 👍 Might've used a `nolint` directive but it's fine regardless

**Pull request merged**